File: erp-backend/xingyun-shkb/src/main/java/com/lframework/xingyun/shkb/enums/TaskType.java
package com.lframework.xingyun.shkb.enums;

import com.baomidou.mybatisplus.annotation.EnumValue;
import com.lframework.starter.web.core.enums.BaseEnum;

/**
 * 任务类型
 */

public enum TaskType implements BaseEnum<String> {
    AVIATION("AVIATION", "民航维修任务"),
    RECEIVE_WB("RECEIVE_WB", "返厂WB任务"),
    RECEIVE_L("RECEIVE_L", "返厂L任务");

    @EnumValue
    private final String code;

    private final String desc;

    TaskType(String code, String desc) {

        this.code = code;
        this.desc = desc;
    }

    @Override
    public String getCode() {

        return this.code;
    }

    @Override
    public String getDesc() {

        return this.desc;
    }
}

File: erp-frontend/src/enums/biz/taskType.ts
import { BaseEnum, BaseEnumItem } from '@/enums/baseEnum';

/**
 * 任务类型枚举
 */
const TASK_TYPE: BaseEnum<string, string> = new BaseEnum<string, string>();
TASK_TYPE.set('AVIATION', new BaseEnumItem<string, string>('AVIATION', '民航维修任务'));
TASK_TYPE.set('RECEIVE_WB', new BaseEnumItem<string, string>('RECEIVE_WB', '返厂WB任务'));
TASK_TYPE.set('RECEIVE_L', new BaseEnumItem<string, string>('RECEIVE_L', '返厂L任务'));

export { TASK_TYPE };
